Source/Pages/Gallery/YPAssetZoomableView.swift

```diff
@@ -277,17 +327,22 @@ fileprivate extension YPAssetZoomableView {
     
     // Centring the image frame
     func centerAssetView() {
-        let assetView = isVideoMode ? videoView : photoImageView
-        let scrollViewBoundsSize = self.bounds.size
-        var assetFrame = assetView.frame
-        let assetSize = assetView.frame.size
-        
-        assetFrame.origin.x = (assetSize.width < scrollViewBoundsSize.width) ?
-            (scrollViewBoundsSize.width - assetSize.width) / 2.0 : 0
-        assetFrame.origin.y = (assetSize.height < scrollViewBoundsSize.height) ?
-            (scrollViewBoundsSize.height - assetSize.height) / 2.0 : 0.0
-        
-        assetView.frame = assetFrame
+        DispatchQueue.main.async { [weak self] in
```

Comment:
I assume one of the call sites is invoking this in a background thread. Could we just wrap the call site in the main thread?